connectors/bitbucket-server/src/main/java/com/vmware/connectors/bitbucket/server/BitBucketServerController.java

```diff
@@ -0,0 +1,435 @@
+                .subscribe(entity -> {
+                    jsonDocuments[0] = entity.getBody();
+                });
+        Assert.isTrue(Objects.nonNull(jsonDocuments[0]), "JsonDocument should not be null.");
+
+        final List<String> comments = jsonDocuments[0].read("$.values[*].comment.text");
+        return comments.stream()
+                .limit(COMMENTS_SIZE)
+                .collect(Collectors.toList());
+    }
+
+    private Single<ResponseEntity<String>> performBitBucketServerAction(final String baseUrl,
+                                                                        final String authHeader,
+                                                                        final BitBucketServerPullRequest pullRequest,
+                                                                        final BitBucketServerAction bitBucketServerAction) {
+        final HttpHeaders headers = buildHeaders(authHeader);
+
+        // Get current version of the pull request. Pull request "version" changes when we do any actions on it.
+        // When the pull request is raised, the current value will be 0.
+        // For example, when we approve the pull request, then the version will change from 0 to 1.
+        // We have to add the latest version of the pull request URI to do any actions. Otherwise, the action will be rejected.
+        // If the build for the branch is going on, then the actions would be rejected.
+        final String version = getVersion(headers, baseUrl, pullRequest);
+
+        final String url = String.format(BitBucketServerConstants.BITBUCKET_PR_ACTION_FORMAT,
+                baseUrl,
+                pullRequest.getProjectKey(),
+                pullRequest.getRepositorySlug(),
+                pullRequest.getPullRequestId(),
+                bitBucketServerAction.getAction(),
+                version);
+
+        // This header is added for skipping the CSRF check. Otherwise, Atlassian denies the pull request action.
+        headers.add(BitBucketServerConstants.ATLASSIAN_TOKEN, "no-check");
+
+        final ListenableFuture<ResponseEntity<String>> response = this.rest.exchange(
+                url,
+                HttpMethod.POST,
+                new HttpEntity<>(headers),
+                String.class);
+        return Async.toSingle(response);
+    }
+
+    private HttpHeaders buildHeaders(final String authHeader) {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.add(AUTHORIZATION, authHeader);
+        return headers;
+    }
+
+    private String getVersion(final HttpHeaders headers,
+                              final String baseUrl,
+                              final BitBucketServerPullRequest pullRequest) {
+
+        final ListenableFuture<ResponseEntity<JsonDocument>> response = getPullRequestInfo(headers, pullRequest, baseUrl);
+        final JsonDocument[] jsonDocuments = new JsonDocument[1];
+        Async.toSingle(response)
+                .subscribe(entity -> {
```

Comment:
My other comment about calling `subscribe` applies here too.

Reply:
I have done the changes to remove subscribe. I have not addressed all the pull requests comments. I have pushed only some of the changes. Apologies for the inconvenience caused.

Reply:
I have done the changes per @RyanBard suggestion.